pkg/cmd/pr/pr.go

```diff
@@ -321,29 +321,28 @@ func (o *Options) SetChangeLog(addChangeLog string) error {
 }
 
 // SetCommitDetails discovers the git URL, and sets the application name, commit message and title
-func (o *Options) SetCommitDetails(dir, commitMessage, commitTitle, application string) error {
-	if commitMessage == "" || commitTitle == "" || application == "" {
-		if application == "" || commitMessage == "" {
+func (o *Options) SetCommitDetails(dir string) error {
+	if o.CommitMessage == "" || o.CommitTitle == "" || o.Application == "" {
+		if o.Application == "" || o.CommitMessage == "" {
 			{
 				gitURL, err := gitdiscovery.FindGitURLFromDir(dir, true)
 				if err != nil {
 					log.Logger().Warnf("failed to find git URL %s", err.Error())
 				} else if gitURL != "" {
-					if application == "" {
+					if o.Application == "" {
 						gitURLPart := strings.Split(gitURL, "/")
 						o.Application = gitURLPart[len(gitURLPart)-2] + "/" +
 							strings.TrimSuffix(gitURLPart[len(gitURLPart)-1], ".git")
 					}
-					if commitMessage == "" {
+					if o.CommitMessage == "" {
 						o.CommitMessage = fmt.Sprintf("from: %s\n", gitURL)
 					}
-				}
-
-				if commitTitle == "" {
-					if application == "" {
-						o.CommitTitle = fmt.Sprintf("chore(deps): upgrade to version %s", o.Version)
-					} else {
-						o.CommitTitle = fmt.Sprintf("chore(deps): upgrade %s to version %s", o.Application, o.Version)
+					if o.CommitTitle == "" {
```

Comment:
You should revert further. Having `if o.CommitTitle == "" {` in this if statement breaks the case when --app is set, but not --commit-title.